forked-daapd: init at 27.2

flyfloh opened the pull request:
NixOS is missing a package for forked-daapd, which is a DAAP (iTunes) and MPD media server with support for AirPlay devices (multiroom), Apple Remote (and compatibles), Chromecast, Spotify and internet radio.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

prusnak commented on pkgs/servers/forked-daapd/default.nix:
```diff
@@ -0,0 +1,90 @@
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    homepage = http://ejurgensen.github.io/forked-daapd;
```
Please put the URL in the `homepage` field into quotes (RFC45).

Infinisil commented:
This PR has way too many options. Check out https://github.com/NixOS/rfcs/pull/42

flyfloh commented:
> This PR has way too many options. Check out https://github.com/NixOS/rfcs/pull/42

Unfortunately I found the RFC repo only after opening this PR. I‘ll fix and update this soon.

flyfloh commented:
I've now:

 * rewritten the module to the style described in RFC42
 * made spotify support optional (it relies on the non-free libspotify)
 * added a test that tries to start the service and checks whether something is reachable on the port

Unfortunately checks are failing in the pipeline. Can someone help me understand what it is that fails?

flyfloh committed:
forked-daapd: init at 27.2

flyfloh committed:
nixos/modules: introduce forked-daapd module

flyfloh committed:
nixos/tests: add test for forked-daapd module

flyfloh commented:
I found the problem and updated the MR. This also now uses the latest version of `forked-daapd`, which allows to specify ports for airplay and can thus run with the firewall enabled.

prusnak commented on pkgs/servers/forked-daapd/default.nix:
```diff
@@ -0,0 +1,89 @@
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    homepage = "http://ejurgensen.github.io/forked-daapd";
```
```suggestion
    homepage = "https://ejurgensen.github.io/forked-daapd";
```

flyfloh replied:
Applied.

prusnak commented on pkgs/servers/forked-daapd/default.nix:
```diff
@@ -0,0 +1,89 @@
+  spotifySupport ? false, libspotify ? null
+}:
+
+let
+  version = "27.2";
+  name = "forked-daapd";
+in stdenv.mkDerivation {
+  inherit name;
```
```suggestion
stdenv.mkDerivation {
  pname = "forked-daapd";
  version = "27.2";
```

ryantm replied:
It also needs to be a recursive attrset:

```
stdenv.mkDerivation rec {
```

flyfloh replied:
Thanks. That works.

prusnak commented on pkgs/servers/forked-daapd/default.nix:
```diff
@@ -0,0 +1,89 @@
+  inherit name;
+
+  src = fetchFromGitHub {
+    repo = name;
```
```suggestion
    repo = pname;
```

flyfloh replied:
I may be holding this wrong but these suggestions give me:

```
error: undefined variable 'pname' at /home/flo/nixpkgs/pkgs/servers/forked-daapd/default.nix:39:12
```

blitz commented on nixos/tests/forked-daapd.nix:
```diff
@@ -0,0 +1,44 @@
+      {
+        services.forked-daapd = {
+          enable = true;
+          libraryPaths = [ /media/music ];
```
`/media/music` should not be imported into the store. I guess this should be `"/media/music"`.

flyfloh replied:
Fixed

blitz commented on nixos/tests/forked-daapd.nix:
```diff
@@ -0,0 +1,44 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+
+let
+  configDir = "/var/lib/foobar";
```
`foobar` is not a helpful name. I think you only specify it to refer to it later, but it would still be nice to have something like `"/var/lib/forked-daapd"` here.

flyfloh replied:
Fixed

blitz reviewed (changes requested):
Looks good. The only thing that IMHO needs changing is the bare pathname.

blitz commented on nixos/tests/forked-daapd.nix:
```diff
@@ -0,0 +1,44 @@
+import ./make-test-python.nix ({ pkgs, ... }:
```
I'm not sure whether this is something that needs fixing:

https://github.com/NixOS/nixpkgs/pull/85550/checks?check_run_id=1432921834

@tfc Any idea?

tfc replied:
log says

```
gcc: error: unrecognized command line option '-m64'
make[1]: *** [Makefile:453: antlr3baserecognizer.lo] Error 1
make[1]: Leaving directory '/build/libantlr3c-3.4'
make: *** [Makefile:308: all] Error 2
error: --- Error --- nix-daemon
builder for '/nix/store/g93kbabwq81cx2r7hcpbvc7cy69qc0h4-libantlr3c-3.4.drv' failed with exit code 2; last 10 log lines:
  building
  build flags: SHELL=/nix/store/93xwsgby6fchr8a58ckgcm2lf4nkv6fl-bash-4.4-p23/bin/bash
  make  all-am
  make[1]: Entering directory '/build/libantlr3c-3.4'
  /nix/store/93xwsgby6fchr8a58ckgcm2lf4nkv6fl-bash-4.4-p23/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -Iinclude    -m64  -O2  -Wall -c -o antlr3baserecognizer.lo `test -f 'src/antlr3baserecognizer.c' || echo './'`src/antlr3baserecognizer.c
  libtool: compile:  gcc -DHAVE_CONFIG_H -I. -Iinclude -m64 -O2 -Wall -c src/antlr3baserecognizer.c  -fPIC -DPIC -o .libs/antlr3baserecognizer.o
  gcc: error: unrecognized command line option '-m64'
  make[1]: *** [Makefile:453: antlr3baserecognizer.lo] Error 1
  make[1]: Leaving directory '/build/libantlr3c-3.4'
  make: *** [Makefile:308: all] Error 2
```

Seems like it's not the test but `libantlr`'s build process
